cityofaustin/atd-moped
# Adds council districts layer to Moped DB

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/10891

This PR gets council districts into our database. We'll definitely need to optimize our queries and make use of indexes if we want to [add council district to the project list](https://github.com/cityofaustin/atd-data-tech/issues/10889). But that's a problem for another day, I hope.


## Testing
**URL to test:** local

**Steps to test:**
1. Spin up the hasura cluster

2. Follow the steps in [the README](https://github.com/cityofaustin/atd-moped/blob/ab72ceb780a3830c467431879d5e962c8b72a36a/moped-toolbox/council_districts_load/README.md) to run both scripts. I've left a copy of `data/council_districts.geojson` in the repo. It seems like we might as well save it, right?

3. You can inspect `data/council_districts.geojson` via https://geojson.io/

4. Open the hasura console and observe the 10 records in the `layer_council_district` table.

5. Create a new project with components spread out across the city.

6. Run this query in your SQL client—it will return **one row per council district** that a project intersects with.

```sql
WITH projects AS (
	SELECT
		plv.*,
		project_geography.feature_id,
		project_geography.geography
	FROM
		project_geography
	LEFT JOIN project_list_view plv ON plv.project_id = project_geography.project_id
)
SELECT
	*
FROM
	projects,
	layer_council_district
WHERE
	ST_Intersects(projects.geography, layer_council_district.geography);
```

7. You'll need to eyeball these results against a [council district map](https://maps.austintexas.gov/GIS/CouncilDistrictMap/)

---
#### Ship list
